runtime/src/iree/hal/drivers/cuda/registration/driver_module.c

```diff
@@ -39,6 +39,34 @@ static iree_status_t iree_hal_cuda_driver_factory_enumerate(
   return iree_ok_status();
 }
 
+#if IREE_HAL_DRIVER_CUDA_NCCL
+static iree_status_t iree_hal_cuda_init_nccl_rank_and_count(
+    iree_hal_cuda_device_params_t* params) {
+  char* nprocs_str = getenv("IREE_SPMD_NPROCS");
```

Comment:
SPMD is too generic - these environment variables should be namespaced and specific to what this is - `IREE_CUDA_NCCL_NPROCS` etc (for this and all the others)